sla_per_use_system_test.py

```diff
@@ -0,0 +1,394 @@
+                               'service_level': ServiceLevel(session=session, name='sla%d' % share, service_shares=share)})
+
+        expected_shares_ratio = self.calculate_shares_ratio(read_users[1]['service_level'].service_shares,
+                                                            read_users[0]['service_level'].service_shares)
+
+        self.log.debug('============== Start {} test'.format(test_name))
+        self.create_auths(entities_list_of_dict=read_users)
+
+        read_cmds = [self.STRESS_READ_CMD.format(n=self.n, user=read_users[0]['user'].name,
+                                                 password=read_users[0]['user'].name,
+                                                 pop='seq=1..%d' % (self.n),
+                                                 duration='%dm' % stress_duration, threads=250),
+                     self.STRESS_READ_CMD.format(n=self.n, user=read_users[1]['user'].name,
+                                                 password=read_users[1]['user'].name,
+                                                 pop='seq=1..%d' % (self.n),
+                                                 duration='%dm' % stress_duration, threads=250)
+                     ]
+
+        try:
+            start_time = time.time()
+
+            results = self.run_stress_get_results(read_cmds=read_cmds,
+                                                  users=read_users,
+                                                  waited_cpu_min=self.WAITED_CPU_MIN)
+            end_time = time.time()
+
+            self.validate_scheduler_runtime(start_time=start_time, end_time=end_time,
+                                            read_users=read_users, expected_ratio=expected_shares_ratio)
+
+            self.assertTrue(results, msg='Not received cassandra-stress results')
+
+            self.log.debug('Validate cassandra-stress ops deviation')
+            actual_shares_ratio = self.calculate_shares_ratio(results[read_users[1]['user'].name],
+                                                              results[read_users[0]['user'].name])
+            self.validate_deviation(expected_ratio=expected_shares_ratio,
+                                    actual_ratio=actual_shares_ratio, msg='Validate cassandra-stress ops.')
+
+            self.log.debug('============== Finish {} test'.format(test_name))
+        except Exception as e:
+            self.log.error(e)
+            error = e.message
+        finally:
+            self.clean_auth(entities_list_of_dict=read_users)
+            return error
+
+    def throughput_latency_load(self, session):
+        """
+        Troughput - latency test:
+        - Add SLA and grant to user (before any load)
+        - user190 with 190 shares
+        - user950 qith 950 shares
+        - Each user runs load from own loader (round robin):
+           - user190 runs load with throttle
+           - user950 runs load with high throughput
+        """
```

Comment:
Is it from cache? from disk? both?
what is the expected result?

Reply:
It's both: disk and cache.
We told about expected result. For now we don't know what is expected result and was decided that I'll create the test that'll run the flow and save the screenshots and snapshots. How to validate this test we'll decide later

Reply:
disk and cahce together? or just cahce at the begining and just disk afterwards? (mention it in the code).
We need a test scenario per case, only-cache, only-disk (harder to reach but possible) and mix.

Regarding Expected:
1. You can add it as TBD and add the relevant code as sample if to fail or pass.
2. We said we will decide based on the results you'll have, you got result, so need to ping the thread again and get answers from Avi/Eliran.